backend/cmd/headlamp.go

```diff
@@ -719,11 +724,54 @@ func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
 	})
 }
 
-func handleClusterAPI(c *HeadlampConfig, router *mux.Router) {
+func handleClusterAPI(c *HeadlampConfig, router *mux.Router) { //nolint:funlen
 	router.PathPrefix("/clusters/{clusterName}/{api:.*}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var contextKey string
 		clusterName := mux.Vars(r)["clusterName"]
 
-		kContext, err := c.kubeConfigStore.GetContext(clusterName)
+		// checking if kubeConfig exists, if not check if the request headers for kubeConfig information
+		kubeConfig := r.Header.Get("KUBECONFIG")
+
+		if kubeConfig != "" && c.enableDynamicClusters {
```

Comment:
the enableDynamicClusters check is not required here as it is already checked
https://github.com/headlamp-k8s/headlamp/blob/1bb8e942e75f0ef843d30ea3be4d9e6b2e6c9049/backend/cmd/headlamp.go#L926